### ISSUE TITLE
Problem with xls writer for conditional style

### ISSUE THREAD

**Issue by A-contresens** [PHPOffice/PhpSpreadsheet#330]
This is:

- [x] a bug report

### What is the expected behavior?

Have same result for xlsx and xls writter

### What is the current behavior?

This code works well with xlsx but does not works with the xls writer.

The conditional rules are applied but they doesn't have the colors information. The bold are ok.

### What are the steps to reproduce?

```php
<?php

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Worksheet\Worksheet;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx;
use PhpOffice\PhpSpreadsheet\IOFactory;
use PhpOffice\PhpSpreadsheet\Style\Color;
use PhpOffice\PhpSpreadsheet\Style\Conditional;
use PhpOffice\PhpSpreadsheet\Style\Fill;

$format = "xls";

$spreadsheet = new Spreadsheet();
$sheet = $spreadsheet->getActiveSheet();

$climateColors = array(
    "1" => "FF6666",
    "2" => "FF9966",
    "3" => "FFFF99",
    "4" => "99CC99",
    "5" => "66CC66",
);

foreach ($climateColors as $climateScore => $color) {
    $conditional = new Conditional();
    $conditional->setConditionType(Conditional::CONDITION_CELLIS)
    ->setOperatorType(Conditional::OPERATOR_EQUAL)
    ->addCondition($climateScore);
    $conditional->getStyle()->getFont()->getColor()->setARGB("FF".$color);
    $conditional->getStyle()->getFont()->setBold(true);
    $conditional->getStyle()->applyFromArray(
        ['fill' => [
                    'fillType' => Fill::FILL_SOLID,
                    'color' => ['argb' => "FF".$color],
                ],
            ]
    );
    $conditionalStyles[] = $conditional;
}

$sheet->setCellValue("A1",1);
$sheet->setCellValue("B1",2);
$sheet->setCellValue("C1",3);
$sheet->setCellValue("D1",4);
$sheet->setCellValue("E1",5);

$spreadsheet->getActiveSheet()->getStyle('A1:E1')->setConditionalStyles($conditionalStyles);

if($format == "xlsx"){
    $writer = new Xlsx($spreadsheet);
} else if ($format == "xls"){
    $writer = IOFactory::createWriter($spreadsheet, 'Xls');
}

$writer->save("/data/tmp/export/test.".$format);
```
 ### Which versions of PhpSpreadsheet and PHP are affected?

php 7.1.4
PhpSpreadsheet 1.0.0

**Comment by PowerKiKi** [comment]
XLS format does not support custom colors, but only a few predefined one. From your list only `FFFF99` is supported. That's a first issue that you might want to fix on your side.

But I can still confirm the bug for supported colors, with the following code:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Writer\Xls;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx;
use PhpOffice\PhpSpreadsheet\Style\Conditional;
use PhpOffice\PhpSpreadsheet\Style\Fill;

$spreadsheet = new Spreadsheet();
$sheet = $spreadsheet->getActiveSheet();

$conditional = new Conditional();
$conditional->setConditionType(Conditional::CONDITION_CELLIS)
    ->setOperatorType(Conditional::OPERATOR_EQUAL)
    ->addCondition('1')
    ->getStyle()->applyFromArray(
        [
            'font' => [
                'bold' => true,
                'color' => ['argb' => "FF0000FF"],
            ],
            'fill' => [
                'fillType' => Fill::FILL_SOLID,
                'color' => ['argb' => "FFFF0000"],
            ],
        ]
    );

$sheet->getStyle('A1')->setConditionalStyles([$conditional]);
$sheet->setCellValue('A1', 1);

$writerXlsx = new Xlsx($spreadsheet);
$writerXls = new Xls($spreadsheet);

$writerXlsx->save("test.xlsx");
$writerXls->save("test.xls");
```

**Comment by oleibman** [comment]
Should be fixed by PR #4030 and PR #4033.
